### PR TITLE
add swap serial test for pull requests

### PR DESCRIPTION
**Pull request by pacoxu**
add test for https://github.com/kubernetes/kubernetes/pull/105271 which is `System reserved swap [Serial]`

### CODE REVIEW

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml [hunk 1]
```diff
@@ -1499,3 +1499,91 @@ presubmits:
...
+            - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
+            - "--service-account=/etc/service-account/service-account.json"
+            - "--upload=gs://kubernetes-jenkins/pr-logs"
+            - "--timeout=240"
+            - --scenario=kubernetes_e2e
+            - --
+            - --deployment=node
+            - --gcp-zone=us-west1-b
+            - --node-args=--image-config-file=/workspace/test-infra/jobs/e2e_node/swap/image-config-swap.yaml
+            - --node-test-args=--feature-gates=NodeSwap=true --kubelet-flags="--cgroups-per-qos=true --cgroup-root=/ --fail-swap-on=false"
```
Comment (mmiranda96) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1530]: In order for the job to properly run, I believe you'll need additional `node-test-args`. I added these on the periodic job: https://github.com/kubernetes/test-infra/pull/24757/files#diff-25d877d3983e5b52e44eb00ff5e6285000bafed96042351a0fcd2875326a61eeR1217

**Review by mmiranda96 — CHANGES_REQUESTED** [review]

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml [hunk 2]
```diff
@@ -1499,3 +1499,91 @@ presubmits:
           requests:
             cpu: 4
             memory: 6Gi
+  - name: pull-kubernetes-node-swap-ubuntu-serial
+    cluster: k8s-infra-prow-build
+    always_run: false
+    optional: true
+    skip_report: false
+    skip_branches:
+      - release-\d+\.\d+  # per-release image
```
Comment (SergeyKanzhelev) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1508]: why skipping release branches? might be needed for cherry-picks going forward

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml (hunk 1 above)
Comment (pacoxu) [reply config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1530]:
```suggestion
            - --node-test-args=--container-runtime=remote --container-runtime-endpoint=unix:///run/containerd/containerd.sock --container-runtime-process-name=/usr/bin/containerd --container-runtime-pid-file= --feature-gates=NodeSwap=true --kubelet-flags="--cgroups-per-qos=true --cgroup-root=/ --fail-swap-on=false --runtime-cgroups=/system.slice/containerd.service" --extra-log="{\"name\": \"containerd.log\", \"journalctl\": [\"-u\", \"containerd*\"]}"'
```

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml [hunk 3]
```diff
@@ -1499,3 +1499,87 @@ presubmits:
...
+            - "--timeout=240"
+            - --scenario=kubernetes_e2e
+            - --
+            - --deployment=node
+            - --gcp-zone=us-west1-b
+            - --node-args=--image-config-file=/workspace/test-infra/jobs/e2e_node/swap/image-config-swap.yaml
+            - --node-test-args=--container-runtime=remote --container-runtime-endpoint=unix:///run/containerd/containerd.sock --container-runtime-process-name=/usr/bin/containerd --container-runtime-pid-file= --feature-gates=NodeSwap=true --kubelet-flags="--cgroups-per-qos=true --cgroup-root=/ --fail-swap-on=false --runtime-cgroups=/system.slice/containerd.service" --extra-log="{\"name\": \"containerd.log\", \"journalctl\": [\"-u\", \"containerd*\"]}"
+            - --node-tests=true
+            - --provider=gce
+            - --test_args=--nodes=8 --focus="\[Serial\]" --skip="\[Flaky\]|\[Slow\]"
```
Comment (ehashman) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1531]: These are serial tests so we shouldn't have 8 nodes. Args should match https://github.com/kubernetes/test-infra/blob/cc95789ad64e7e0234b418781d4f01cac31928cd/config/jobs/kubernetes/sig-node/node-kubelet.yaml#L11-L24

but we should use the swap image config and have the swap feature flag enabled.

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml [hunk 4]
```diff
@@ -1499,3 +1499,87 @@ presubmits:
...
+        - "--timeout=240"
+        - --scenario=kubernetes_e2e
+        - -- # end bootstrap args, scenario args below
+        - --deployment=node
+        - --env=KUBE_SSH_USER=core
+        - --gcp-zone=us-west1-b
+        - '--node-test-args=--feature-gates=NodeSwap=true --container-runtime=remote --container-runtime-endpoint=unix:///var/run/crio/crio.sock --container-runtime-process-name=/usr/local/bin/crio --container-runtime-pid-file= --kubelet-flags="--fail-swap-on=false --cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/crio.service --kubelet-cgroups=/system.slice/kubelet.service --non-masquerade-cidr=0.0.0.0/0" --extra-log="{\"name\": \"crio.log\", \"journalctl\": [\"-u\", \"crio\"]}"'
+        - --node-tests=true
+        - --provider=gce
+        - --test_args=--nodes=8 --focus="\[Serial\]" --skip="\[Flaky\]|\[Slow\]"
```
Comment (ehashman) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1576]: Same feedback as above, see https://github.com/kubernetes/test-infra/blob/cc95789ad64e7e0234b418781d4f01cac31928cd/config/jobs/kubernetes/sig-node/node-kubelet.yaml#L50-L64 for serial config

**Review by ehashman — CHANGES_REQUESTED** [review]

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml (hunk 3 above)
Comment (pacoxu) [reply config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml]: I changed the job following your example. 

The only question is that why we use `/workspace/test-infra/jobs/e2e_node/containerd/image-config-serial.yaml` not `image-config-swap-fedora.yaml` for Swap serial tests.

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml (hunk 4 above)
Comment (pacoxu) [reply config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml]: Updated

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml [hunk 5]
```diff
@@ -1499,3 +1499,87 @@ presubmits:
...
+            - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
+            - "--service-account=/etc/service-account/service-account.json"
+            - "--upload=gs://kubernetes-jenkins/pr-logs"
+            - --timeout=260
+            - --root=/go/src
+            - --scenario=kubernetes_e2e
+            - --
+            - --deployment=node
+            - --gcp-zone=us-west1-b
+            - --node-args=--image-config-file=/workspace/test-infra/jobs/e2e_node/containerd/image-config-serial.yaml
```
Comment (endocrimes) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1527]: There's no swap configured on the node by this config, so we don't really get to see any breakages caused by the presence of swap

Comment (pacoxu) [reply config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1527]: https://github.com/kubernetes/test-infra/pull/24919#discussion_r794167018 
I have the question as well. I will update the pr. Just get back from a holiday.

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml (hunk 4 above)
Comment (ehashman) [reply config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1580]: Node count is still set incorrectly, we also need to update skipped tests.

**Review by ehashman — CHANGES_REQUESTED** [review]

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml [hunk 6]
```diff
@@ -1499,3 +1499,91 @@ presubmits:
...
+        - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
+        - "--service-account=/etc/service-account/service-account.json"
+        - "--upload=gs://kubernetes-jenkins/pr-logs"
+        - "--timeout=240"
+        - --scenario=kubernetes_e2e
+        - -- # end bootstrap args, scenario args below
+        - --deployment=node
+        - --env=KUBE_SSH_USER=core
+        - --gcp-zone=us-west1-b
+        - '--node-test-args=--feature-gates=NodeSwap=true --container-runtime=remote --container-runtime-endpoint=unix:///var/run/crio/crio.sock --container-runtime-process-name=/usr/local/bin/crio --container-runtime-pid-file= --kubelet-flags="--fail-swap-on=false --cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/crio.service --kubelet-cgroups=/system.slice/kubelet.service --non-masquerade-cidr=0.0.0.0/0" --extra-log="{\"name\": \"crio.log\", \"journalctl\": [\"-u\", \"crio\"]}"'
```
[the remark this answers is not in the document]
Comment (pacoxu) [reply config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml]: Updated.

**Review by ehashman — COMMENTED** [review]
/lgtm
/approve

let's try it and fix anything else in a follow-up. if this looks good let's add periodics as well

**Comment by pacoxu** [comment]
- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105271/pull-kubernetes-node-swap-fedora-serial/1492680680927137792/
- https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-node-swap-ubuntu-serial

jobs failed for below errors.
> W0213 02:07:56.767] E0213 02:07:56.767641    6858 ssh.go:123] failed to run SSH command: out: core@34.82.165.28: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
W0213 02:07:56.768] , err: exit status 255

**Pull request merged**